dropwizard/src/main/java/com/linecorp/armeria/server/dropwizard/ArmeriaServerFactory.java

```diff
@@ -0,0 +1,311 @@
+                                             ArmeriaServerDecorator.class.getName());
+            }
+            ((ArmeriaServerDecorator) connectorFactory).decorate(sb);
+        }
+        if (meterRegistry != null) {
+            sb.meterRegistry(meterRegistry);
+        }
+        if (writerFactory != null && !writerFactory.getWriter()
+                                                   .equals(AccessLogWriter.disabled())) {
+            LOGGER.trace("Setting up Armeria AccessLogWriter");
+            sb.accessLogWriter(writerFactory.getWriter(), true);
+        } else {
+            LOGGER.info("Armeria access logs will not be written");
+            sb.accessLogWriter(AccessLogWriter.disabled(), true);
+        }
+        return sb;
+    }
+
+    @JsonProperty
+    private @Valid ConnectorFactory connector = ArmeriaHttpConnectorFactory.build();
+    @JsonProperty
+    private @Valid @NotNull AccessLogWriterFactory accessLogWriter = new CommonAccessLogWriterFactory();
+    @JsonProperty
+    private boolean jerseyEnabled = true;
+    @JsonProperty
+    private @MinSize(0) Size maxRequestLength = Size.bytes(Flags.defaultMaxRequestLength());
+    @JsonProperty
+    private boolean disableDateHeader;
+    @JsonProperty
+    private boolean disableServerHeader;
+    @JsonProperty
+    private boolean verboseResponses;
+    @JsonProperty
+    @Nullable
+    private String defaultHostname;
+    @JsonIgnore
+    private transient ServerBuilder serverBuilder;
+
+    public ArmeriaServerFactory() {
+    }
+
+    /**
+     * Sets up the Armeria ServerBuilder with values from the Dropwizard Configuration.
+     * Ref <a href="https://line.github.io/armeria/advanced-production-checklist.html">Production Checklist</a>
+     *
+     * @param serverBuilder A non-production ready ServerBuilder
+     * @return A production-ready ServerBuilder
+     */
+    // visible for testing
+    ServerBuilder decorateServerBuilderFromConfig(final ServerBuilder serverBuilder) {
+        Objects.requireNonNull(serverBuilder);
+        serverBuilder.maxNumConnections(getMaxQueuedRequests());
+        serverBuilder.blockingTaskExecutor(Executors.newFixedThreadPool(getMaxThreads()), true);
+        serverBuilder.maxRequestLength(maxRequestLength.toBytes());
+        serverBuilder.idleTimeoutMillis(getIdleThreadTimeout().toMilliseconds());
+        serverBuilder.gracefulShutdownTimeout(
+                Duration.ofSeconds(30L),
+                Duration.ofMillis(getShutdownGracePeriod().toMilliseconds()));
+        if (isDateHeaderDisabled()) {
+            serverBuilder.disableDateHeader();
+        }
+        if (isServerHeaderDisabled()) {
+            serverBuilder.disableServerHeader();
+        }
+        serverBuilder.verboseResponses(hasVerboseResponses());
+        Optional.ofNullable(getDefaultHostname()).ifPresent(serverBuilder::defaultHostname);
+        // TODO: Add more items to server builder via Configuration
```

Comment:
You might want to set the `MeterRegistry` of the server:

```java
serverBuilder.meterRegistry(DropwizardMeterRegistries.newRegistry(dropwizardMetricsRegistry));
```

Reply:
Alternatively, you could add the `DropwizardMeterRegistry` to `Metrics.globalRegistry`, so the metrics created via Micrometer API are also exported to Dropwizard Metrics:

```java
Metrics.globalRegistry.addRegistry(DropwizardMeterRegistries.newRegistry(...));
// No need to call `serverBuilder.meterRegistry()` because it defaults to `Metrics.globalRegistry`.
```

Reply:
Already set it. 
https://github.com/line/armeria/pull/2236/files/2d5e7a08793cc21d655a114a45f887fc74b957ed#diff-6c4e58dd555e8837e0fc639603cd52ecR105-R107

Reply:
Ah, OK!